core/instrumentation.ts

```diff
@@ -0,0 +1,6 @@
+export async function register() {
+  if (process.env.NEXT_PUBLIC_API_MOCKING === 'enabled' && process.env.NEXT_RUNTIME === 'nodejs') {
+    const { server } = await import('~/client/mocks/server');
+    server.listen();
+  }
+}
```

Comment:
So this doesn't actually work... Went to try this myself and it's not actually running the service worker as we are using the `edge` runtime. We'd also need the `NEXT_PUBLIC_API_MOCKING` environment variable set on the deployment, which I don't think we should do, especially since we run tests on the production deployments (we want users to be able to create accounts to test around with).

I'd also say it'd be alright that we register a bunch of customers in the CP as we already do this for orders right now. We could also pass in a custom provider/domain so we can clean them up eventually: `faker.person.email({ provider: 'mybigcommerce.com' })`

Reply:
Ah, good catch @chanceaclark  - so, this means Mock Service Worker is not actually suitable for us. 
Ok, I will close this PR and create a new one only with register customer test. 

> I'd also say it'd be alright that we register a bunch of customers in the CP as we already do this for orders right now. 

We can do that - looks like we don't have much of a choice at the moment. But before doing that, I think it would be nice to define some data clean up strategy - but it is a matter of a separate discussion, I believe.